// jsxte/dist/esm/html-parser/base-html-parser/base-html-parser.mjs
// src/html-parser/base-html-parser/base-html-parser.ts
import { mapAttributeName } from "../../string-template-parser/map-attribute-name.mjs";
var HTMLElementResolver = class {
  constructor(attributeMap) {
    this.attributeMap = {};
    this.attributeMap = attributeMap;
  }
  resolveAttributes(element) {
    const attributes = [];
    for (const [key, prop] of Object.entries(element.props)) {
      if (key !== "children") {
        attributes.push([mapAttributeName(key, this.attributeMap), prop]);
      }
    }
    return attributes;
  }
  resolveChildren(element) {
    if (!element.props.children)
      return [];
    if (Array.isArray(element.props.children))
      return element.props.children;
    return [element.props.children];
  }
  resolveElement(element) {
    const children = this.resolveChildren(element);
    const attributes = this.resolveAttributes(element);
    return {
      tag: element.tag.toString(),
      children,
      attributes
    };
  }
};
export {
  HTMLElementResolver
};
