// jsxte/dist/cjs/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var src_exports = {};
__export(src_exports, {
  ErrorBoundary: () => import_error_boundary.ErrorBoundary,
  defineContext: () => import_context_map.defineContext,
  memo: () => import_memo.memo,
  renderToHtml: () => import_render_to_html.renderToHtml,
  renderToHtmlAsync: () => import_render_to_html.renderToHtmlAsync,
  renderToStringTemplateTag: () => import_render_to_string_template_tag.renderToStringTemplateTag
});
module.exports = __toCommonJS(src_exports);
var import_array_flat_polyfill = require("./utilities/array-flat-polyfill.cjs");
__reExport(src_exports, require("./express/index.cjs"), module.exports);
__reExport(src_exports, require("./jsx/jsx.types.cjs"), module.exports);
__reExport(src_exports, require("./jsx/prop-types/index.cjs"), module.exports);
var import_error_boundary = require("./error-boundary/error-boundary.cjs");
var import_context_map = require("./context-map/context-map.cjs");
var import_render_to_html = require("./html-parser/render-to-html.cjs");
var import_render_to_string_template_tag = require("./string-template-parser/render-to-string-template-tag.cjs");
var import_memo = require("./utilities/memo.cjs");
